icicle/include/icicle/backend/README.md

```diff
@@ -0,0 +1,76 @@
+# How to Register a New Function
+
+For users to be able to use a backend function (whenever it's a CPU/GPU one), the function should be exposed to the user and be registered to the matching backend. We do not let the user call the backend function directly, instead a wrapping function, with identical signature to all backend, is created as a user-facing function. Then, the backend specific functions are registered to their respective backends in such a way that a dispatcher will know to call the appropriate function (according to the device used) whenever the user facing function is used.
+Here is a step-by-step explanation of how to do that registration process right.
+
+## The backend specific function
+The first step is to write the backend specific function. You should notice that its first argument must be `Device`. For example:
```

Comment:
I want to suggest a different order of things that makes more sense to me. You can ignore if you disagree:
1. first you define a template frontend API (this is used by the user directly and this is all he has to know) and config. **This is done in vec_ops.h for example.**
2. Then you declare the type that the backend needs to register and a register function. **This is in vec_ops_backend.h**
3. Now you need to instantiate the dispatcher (using ICICLE_DISPATCHER_INST) which also defines the register function (from 2). **This must be in vec_ops.cpp since this is compiled**
4. Finally a backend is implementing a function that matches the prototype from step 2 and uses the REGISTER macro to register it.